src/pytorch_lightning/plugins/precision/tpu.py

```diff
@@ -39,8 +46,10 @@ def optimizer_step(  # type: ignore[override]
     ) -> Any:
         import torch_xla.core.xla_model as xm
 
+        closure = partial(self._tpu_wrap_closure, optimizer, closure)
         closure = partial(self._wrap_closure, model, optimizer, optimizer_idx, closure)
-        closure_result = xm.optimizer_step(optimizer, optimizer_args={"closure": closure, **kwargs})
+        closure_result = optimizer.step(closure=closure, **kwargs)
+        xm.mark_step()
```

Comment:
So essentially what we needed was barrier=True here, but to make it work with the closure, you had to pull out the implementation from around the closure: https://github.com/pytorch/xla/blob/b2bd721cf18b496feec02f12313bb532484eaab9/torch_xla/core/xla_model.py#L1030-L1032
Is my explanation correct?

Reply:
Yes, that's right.